"discoverd: timed out waiting for instances" failure on bootstrap

Failure on bootstrap:
```
17:45:48.092460 wait-controller wait-controller
17:46:18.094218 host.go:142: error getting controller instance: discoverd: timed out waiting for instances'
```

https://gist.github.com/5c8fbcd0103f3efe97a537780ba5377a is the debug-info from the boostrapping node.

--- a/controller/migrate_test.go
+++ b/controller/migrate_test.go
@@ -106,13 +106,15 @@ func (MigrateSuite) TestMigrateReleaseArtifacts(c *C) {
 	}
 	c.Assert(db.Exec(`INSERT INTO releases (release_id) VALUES ($1)`, random.UUID()), IsNil)
 
-	// insert a slug based release
-	slugReleaseID := random.UUID()
+	// insert multiple slug based releases with the same slug URI
+	slugReleaseIDs := []string{random.UUID(), random.UUID()}
 	imageArtifactID := random.UUID()
 	slugEnv := map[string]string{"SLUG_URL": "http://example.com/slug.tgz"}
 	c.Assert(db.Exec(`INSERT INTO artifacts (artifact_id, type, uri) VALUES ($1, $2, $3)`, imageArtifactID, "docker", "http://example.com/"+imageArtifactID), IsNil)
-	c.Assert(db.Exec(`INSERT INTO releases (release_id, artifact_id, env) VALUES ($1, $2, $3)`, slugReleaseID, imageArtifactID, slugEnv), IsNil)
-	releaseArtifacts[slugReleaseID] = imageArtifactID
+	for _, id := range slugReleaseIDs {
+		c.Assert(db.Exec(`INSERT INTO releases (release_id, artifact_id, env) VALUES ($1, $2, $3)`, id, imageArtifactID, slugEnv), IsNil)
+		releaseArtifacts[id] = imageArtifactID
+	}
 
 	// migrate to 15 and check release_artifacts was populated correctly
 	m.migrateTo(15)
@@ -128,19 +130,21 @@ func (MigrateSuite) TestMigrateReleaseArtifacts(c *C) {
 	c.Assert(rows.Err(), IsNil)
 	c.Assert(actual, DeepEquals, releaseArtifacts)
 
-	// check the slug release got "git=true" in metadata
-	var releaseMeta map[string]string
-	err = db.QueryRow("SELECT meta FROM releases WHERE release_id = $1", slugReleaseID).Scan(&releaseMeta)
-	c.Assert(err, IsNil)
-	c.Assert(releaseMeta, DeepEquals, map[string]string{"git": "true"})
+	for _, id := range slugReleaseIDs {
+		// check the slug releases got "git=true" in metadata
+		var releaseMeta map[string]string
+		err = db.QueryRow("SELECT meta FROM releases WHERE release_id = $1", id).Scan(&releaseMeta)
+		c.Assert(err, IsNil)
+		c.Assert(releaseMeta, DeepEquals, map[string]string{"git": "true"})
 
-	// check the slug release got a file artifact with the correct URI and meta
-	var slugURI string
-	var artifactMeta map[string]string
-	err = db.QueryRow("SELECT uri, meta FROM artifacts INNER JOIN release_artifacts USING (artifact_id) WHERE type = 'file' AND release_id = $1", slugReleaseID).Scan(&slugURI, &artifactMeta)
-	c.Assert(err, IsNil)
-	c.Assert(slugURI, Equals, slugEnv["SLUG_URL"])
-	c.Assert(artifactMeta, DeepEquals, map[string]string{"blobstore": "true"})
+		// check the slug releases got a file artifact with the correct URI and meta
+		var slugURI string
+		var artifactMeta map[string]string
+		err = db.QueryRow("SELECT uri, meta FROM artifacts INNER JOIN release_artifacts USING (artifact_id) WHERE type = 'file' AND release_id = $1", id).Scan(&slugURI, &artifactMeta)
+		c.Assert(err, IsNil)
+		c.Assert(slugURI, Equals, slugEnv["SLUG_URL"])
+		c.Assert(artifactMeta, DeepEquals, map[string]string{"blobstore": "true"})
+	}
 }
 
 // TestMigrateArtifactMeta checks that migrating to ID 16 correctly

--- a/controller/schema.go
+++ b/controller/schema.go
@@ -321,18 +321,23 @@ $$ LANGUAGE plpgsql`,
 		// set "git=true" for releases with SLUG_URL set
 		`UPDATE releases SET meta = jsonb_merge(CASE WHEN meta = 'null' THEN '{}' ELSE meta END, '{"git":"true"}') WHERE env ? 'SLUG_URL'`,
 
-		// create file artifacts for any releases with SLUG_URL set
+		// create file artifacts for any releases with SLUG_URL set,
+		// taking care not to create duplicate artifacts
 		`DO $$
 		DECLARE
 			release RECORD;
+			artifact uuid;
 		BEGIN
 			FOR release IN SELECT * FROM releases WHERE env ? 'SLUG_URL' LOOP
-				WITH artifact AS (
+				SELECT INTO artifact artifact_id FROM artifacts WHERE type = 'file' AND uri = release.env->>'SLUG_URL';
+
+				IF NOT FOUND THEN
 					INSERT INTO artifacts (type, uri, meta)
 					VALUES ('file', release.env->>'SLUG_URL', '{"blobstore":"true"}')
-					RETURNING *
-				)
-				INSERT INTO release_artifacts (release_id, artifact_id) (SELECT release.release_id, artifact_id FROM artifact);
+					RETURNING artifact_id INTO artifact;
+				END IF;
+
+				INSERT INTO release_artifacts (release_id, artifact_id) VALUES(release.release_id, artifact);
 			END LOOP;
 		END $$`,
 		`ALTER TABLE releases DROP COLUMN artifact_id`,

